### PR TITLE
Fix/improvement pie radius with outer data labels

### PR DESCRIPTION
**Pull request by jungeun-cho**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description
* When calculating the radius of the pie chart, it is affected by the outer data label. (using `anchor: 'outer'`)
* The minimum radius is `10`.
* Changed the `lineWidth` of the pie series to `3` on hover or on selection.
스크린샷 2021-01-21 23 14 07

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: apps/chart/src/helpers/theme.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ const boxplotDefault = {
 };
 
 export const DEFAULT_BULLET_RANGE_OPACITY = [0.5, 0.3, 0.1];
-const DEFAULT_PIE_LINE_WIDTH = 5;
+const DEFAULT_PIE_LINE_WIDTH = 3;
```
Comment (jung-han) [inline apps/chart/src/helpers/theme.ts:42]: ㅋㅋㅋ감사합니다..

File: apps/chart/src/component/pieSeries.ts [hunk 2]
```diff
@@ -109,6 +115,52 @@ function getPieSeriesOpacityByDepth(
...
+      ['00.00%'], // up to 5 digits
+      getFont(theme)
+    );
+
+    percentLabelWidth = maxLabelWidth;
+    percentLabelHeight = maxLabelHeight;
+  }
+
+  let seriesNameLabelWidth = 0;
+  let seriesNameLabelHeight = 0;
```
Comment (jung-han) [inline apps/chart/src/component/pieSeries.ts:140]: 음..선호도의 차이이겠지만

```js
let seriesNameLabelSize = {
  width: 0,
  height: 0,
};

if(...) {
  seriesNameLabelWidth = {
    ...getMaxLabelSize(seriesNameLabels, getFont(theme.pieSeriesName!)),
  }
}
```

는 어떠신가요?(위도 동일합니다.)

File: apps/chart/src/component/pieSeries.ts [hunk 3]
```diff
@@ -182,18 +234,35 @@ export default class PieSeries extends Component {
...
     if (nestedPieSeries) {
       const { data } = nestedPieSeries[this.alias];
 
       const pieAlias = Object.keys(nestedPieSeries);
       const pieIndex = pieAlias.findIndex((alias) => alias === this.alias);
-      const renderOptionsMap = this.getRenderOptionsMap(options, pieAlias);
+
+      // check the data label of the last Pie series
+      const lastAlias = pieAlias[pieAlias.length - 1];
+      const lastSeries = nestedPieSeries[lastAlias];
```
Comment (jung-han) [inline apps/chart/src/component/pieSeries.ts:247]: `last`사용하는건 어떤가요?

File: apps/chart/src/helpers/pieSeries.ts [hunk 4]
```diff
@@ -46,11 +45,23 @@ export function isSemiCircle(clockwise: boolean, startAngle: number, endAngle: n
...
+
+  if (isSemiCircular) {
+    result = Math.min(width / 2, height) - maxDataLabelHeight;
+  } else if (width > height) {
+    result = height / 2 - maxDataLabelHeight;
+  } else {
+    result = width / 2 - maxDataLabelWidth;
+  }
+
+  return Math.max(result, 10);
```
Comment (jung-han) [inline apps/chart/src/helpers/pieSeries.ts:64]: 10은 상수로 두는게 좋을 것 같습니다

**Review by jung-han — APPROVED** [review]
완료입니다!

File: apps/chart/src/component/pieSeries.ts [hunk 5]
```diff
@@ -109,6 +115,52 @@ function getPieSeriesOpacityByDepth(
...
+
+  return {
+    maxDataLabelWidth: hasOuterLabel
+      ? Math.max(percentLabelWidth, seriesNameLabelWidth) + RADIUS_PADDING
+      : 0,
+    maxDataLabelHeight: hasOuterLabel
+      ? Math.max(percentLabelHeight, seriesNameLabelHeight) + RADIUS_PADDING
+      : 0,
+  };
+}
```
Comment (shiren) [inline apps/chart/src/component/pieSeries.ts:162]: seriesNameLabelWidth대신 maxLabelWidth의 값을 만들어가는 형태로 코드가 전개되면 어떨까 싶어요. maxLabelWidth의 초기값으로 0이 아닌 percentLabelWidth로 초기화 하고 hasOuterPieSeriesNameLabel면 Math.max(percentLabelWidth, seriesNameLabelWidth) 다시 할당하는거죠. 리턴할때는 maxLabelWidth을 그냥 사용하는것이구요

**Review by shiren — APPROVED** [review]
수고하셨어요

File: apps/chart/src/component/pieSeries.ts (hunk 5 above)
Comment (jungeun-cho) [reply apps/chart/src/component/pieSeries.ts:162]: 이렇게 바꾸었습니다~!
```ts
  return outerLabels.reduce(
    (acc, cur) => {
      const { width, height } = acc;
      const { hasOuterLabel, labels, theme } = cur;

      if (hasOuterLabel) {
        const { maxLabelWidth, maxLabelHeight } = getMaxLabelSize(labels, getFont(theme));

        return {
          width: Math.max(maxLabelWidth + RADIUS_PADDING, width),
          height: Math.max(maxLabelHeight + RADIUS_PADDING, height),
        };
      }

      return acc;
    },
    { width: 0, height: 0 }
  );
```

**Pull request merged**